include/libtorrent/lazy_entry.hpp

```diff
@@ -378,7 +378,7 @@ namespace libtorrent
 
 		// the number of bytes this entry extends in the
 		// bencoded buffer
-		std::uint32_t m_len;
+		std::int32_t m_len;
```

Comment:
this might be a bit delicate. there used to be a bug in the bdecoder where overflowing a 32 bit and effectively circumvent overflow checks